lightly/data/_video.py

```diff
@@ -112,61 +127,125 @@ def read_frame(self, timestamp = None):
         to the right position and then load the frame.
         
         Args:
-            timestamp: Specific timestamp of frame in seconds or None (default: None)
+            timestamp: 
+                Specific timestamp of frame in seconds or None (default: None)
 
         Returns:
             A PIL Image
 
+        Raises:
+            StopIteration:
+                If video is empty or end of video is reached and timestamp is None.
+            ValueError: 
+                If provided timestamp is not in self.timestamps.
+            RuntimeError:
+                If loaded frame has wrong shape.
+
         """
-        if timestamp is not None:
-            if (
-                self.current_timestamp_idx + 1 < len(self.timestamps)
-                and self.timestamps[self.current_timestamp_idx + 1] == timestamp
-            ):
-                # Timestamp is timestamp of next frame. Set current timestamp
-                # index to next frame.
-                self.current_timestamp_idx += 1
+        if not self.timestamps:
+            # Empty video.
+            raise StopIteration()
```

Comment:
we shouldn't call `StopIteration` in `read_frame` this will stop the dataloader!

Reply:
What should we do instead if the video is empty? A call to `next(self.reader)` will also raise StopIteration.

Reply:
Mid- to long-term we'll probably have to add an `EmptyVideoError` or similar which we can catch.